### PR TITLE
Use streams and remove dead code in CharacterDisplay

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: code/src/java/pcgen/core/display/CharacterDisplay.java [hunk 1]
```diff
@@ -747,15 +744,10 @@ public int getRacialHDSize()
...
-		{
-			if (po.getSafe(ObjectKey.VISIBILITY).isVisibleTo(v))
-			{
-				aList.add(po);
-			}
-		}
-		return aList;
+		return skillFacet.getSet(id)
+		                 .stream()
+		                 .filter(po -> po.getSafe(ObjectKey.VISIBILITY).isVisibleTo(v))
```
Comment (karianna) [inline code/src/java/pcgen/core/display/CharacterDisplay.java]: what's a `po` and a `v`? better var names please.

**Comment by grimreaper** [comment]
also fixed some related issues

**Comment by LegacyKing** [comment]
Looks like you changed how a Unit Test displayed - Please fix.

`pcgen.inttest.game_pathfinder.pcGenGUIPfrpgGoldielocksTest > testCode FAILED
    java.lang.AssertionError: Expected text value 'Base Dragon HD (8), Dragon (Colossal), Dragon Attacks HD (8), Dragon Size HD (8)' but was 'Base Dragon HD (8), Dragon Size HD (8), Dragon Attacks HD (8), Dragon (Colossal)' - comparing <list ...>Base Dragon HD (8), Dragon (Colossal), Dragon Attacks HD (8), Dragon Size HD (8)</list> at /character[1]/templates[1]/list[1]/text()[1] to <list ...>Base Dragon HD (8), Dragon Size HD (8), Dragon Attacks HD (8), Dragon (Colossal)</list> at /character[1]/templates[1]/list[1]/text()[1]
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at pcgen.inttest.PcGenGUITestCase.runTest(PcGenGUITestCase.java:138)
        at pcgen.inttest.game_pathfinder.pcGenGUIPfrpgGoldielocksTest.testCode(pcGenGUIPfrpgGoldielocksTest.java:48)`

**Comment by grimreaper** [comment]
Replacing with a different more full PR

**Pull request closed without merging**